core/services/workflows/workflow.go

```diff
@@ -17,19 +18,105 @@ type stepRequest struct {
 }
 
 // stepDefinition is the parsed representation of a step in a workflow.
+//
+// Within the workflow spec, they are called "Capability Properties".
 type stepDefinition struct {
-	Type   string         `yaml:"type"`
-	Ref    string         `yaml:"ref"`
-	Inputs map[string]any `yaml:"inputs"`
-	Config map[string]any `yaml:"config"`
+	// A universally unique name for a capability will be defined under the “type” property. The uniqueness will, eventually, be enforced in the Capability Registry . Semver must be used to specify the version of the Capability at the end of the type field. Capability versions must be immutable.
+	//
+	// Initially, we will require major versions. This will ease upgrades early on while we develop the infrastructure.
+	//
+	// Eventually, we might support minor version and specific version pins. This will allow workflow authors to have flexibility when selecting the version, and node operators will be able to determine when they should update their capabilities.
+	//
+	// There are two ways to specify a type - using a string as a fully qualified ID or a structured table. When using a table, tags are ordered alphanumerically and joined into a string following a
+	//  {type}:{tag1_key}_{tag1_value}:{tag2_key}_{tag2_value}@{version}
+	// pattern.
+	//
+	// The “type” supports [a-z0-9_-:] characters followed by an @ and [semver regex] at the end.
+	//
+	// Validation must throw an error if:
+	//
+	// Unsupported characters are used.
+	// (For Keystone only.) More specific than a major version is specified.
+	//
+	// Example (string)
+	//  type: read_chain:chain_ethereum:network_mainnet@1
+	//
+	// Example (table)
+	//
+	//  type:
+	//    name: read_chain
+	//    version: 1
+	//    tags:
+	//      chain: ethereum
+	//      network: mainnet
+	//
+	// [semver regex]: https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+	Type string `json:"type" jsonschema:"required,pattern=^[a-z0-9_\\-:]+@(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$"`
+
+	// Actions and Consensus capabilities have a required “ref” property that must be unique within a Workflow file (not universally) This property enables referencing outputs and is required because Actions and Consensus always need to be referenced in the following phases. Triggers can optionally specify  if they need to be referenced.
+	//
+	// The “ref” supports [a-z0-9_] characters.
+	//
+	// Validation must throw an error if:
+	//	- Unsupported characters are used.
```

Comment:
nit: spacing

Reply:
weird, my editor is mixing spaces and tabs when it should not be